docs/en/sql-reference/functions/ext-dict-functions.md

```diff
@@ -481,44 +443,1954 @@ SELECT dictGetAll('regexp_dict', 'tag', 'foobarbaz', 2);
-- `default_value_expr` — Value returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) returning the value in the data type configured for the `attr_name` attribute.
+- `dictionary_name`: the name of the dictionary you want to retrieve data from. [String literal](../syntax#syntax-string-literal)
+- `attribute_name`: the attribute whose value you want to get. [String literal](../syntax#syntax-string-literal), or [tuple](../data-types/tuple) of column names.
+- `key`: the key associated with the attribute. [Expression](../syntax#syntax-expressions) returning dictionary key-type value.
+- `default_value_expr` — Values returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) or [Tuple](../../sql-reference/data-types/tuple.md)([Expression](../../sql-reference/syntax.md#syntax-expressions)), returning the value (or values) in the data types configured for the `attr_names` attribute.
 
 **Returned value**
 
-- If ClickHouse parses the attribute successfully in the [attribute’s data type](../../sql-reference/dictionaries/index.md#dictionary-key-and-fields#ext_dict_structure-attributes), functions return the value of the dictionary attribute that corresponds to `id_expr`.
+The value of the dictionary attribute that corresponds to the given key. This value is parsed and returned as an `Int8` type. If the attribute cannot be found, the value of `default_value_expr` is returned instead.
 
-- If there is no requested `id_expr` in the dictionary then:
+**Implementation details**
 
-        - `dictGet[Type]` returns the content of the `<null_value>` element specified for the attribute in the dictionary configuration.
-        - `dictGet[Type]OrDefault` returns the value passed as the `default_value_expr` parameter.
+This function will throw an exception if the key does not exist in the dictionary.
 
-ClickHouse throws an exception if it cannot parse the value of the attribute or the value does not match the attribute data type.
+**Examples**
+
+Query:
+
+```sql
+CREATE TABLE countries
+(
+    id UInt64,
+    country String,
+    code_designation Int8
+) ENGINE = MergeTree
+ORDER BY id;
+
+INSERT INTO countries VALUES (1, 'Mali', 32);
+INSERT INTO countries VALUES (2, 'Czechia', 28);
+INSERT INTO countries VALUES (3, 'Brazil', 82);
+
+CREATE DICTIONARY country_codes
+(
+    id UInt64,
+    country String,
+    code_designation Int8
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'countries'))
+LAYOUT(FLAT())
+LIFETIME(MIN 0 MAX 1000);
+
+SELECT dictGetInt8OrDefault('country_codes', 'code_designation', '999', 'Default return string.') AS code_designation_output;
+```
+
+Result:
+
+```response
+'Default return string.'
+```
+
+## dictGetInt16
```

Comment:
Do we really need this copy paste for every function `dictGet<someType>`? I don't think it's necessary, the are all the same and we can make some general documentation for `dictGet<someType>` and `dictGet<someType>OrDefault`, we can just list supported types and add several examples

Reply:
Search engine optimization.

Reply:
What optimizations? I don't see why we need huge copy paste in docs with the same text and examples

Reply:
If a user searches for `dictGetInt16`, that means they're looking for documentation for the function `dictGetInt16`. Not Int8, not Float32, not anything else. They know what they want, and what they want is `dictGtiInt16`.

In the future, we can automate some of this documentation to populate the changes for us. But right now, the docs team's goal is to document all the holes left by engineers of old. That's what we're doing.

Reply:
I still think if the user searches for `dictGetInt16` and finds general documentation about `dictGet<type_name>`, it will be enough to understand everything about `dictGetInt16`. For me having 2k lines of documentation for each `dictGet<type_name>` is not a good documentation. But I can be wrong, let me ask what my colleagues think about it.